numpy/core/src/multiarray/sequence.c

```diff
@@ -50,9 +50,19 @@ array_contains(PyArrayObject *self, PyObject *el)
     return ret;
 }
 
+static PyObject*
+array_concat(PyObject *self, PyObject *other) {
+    // Throw a type error, when trying to concat NDArrays
+    if(PyArray_Check(other) || PyArray_Check(self)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "Concatenation operation is not implemented for Numpy Arrays");
```

Comment:
Maybe it's worth highlighting `np.concatenate` in the error message? Only as long as it doesn't make things too verbose.

Reply:
Agreed, makes sense to redirect